src/meltano/core/plugin/singer/tap.py

```diff
@@ -615,3 +618,56 @@ def catalog_cache_key(self, plugin_invoker):
         key_json = json.dumps(key_dict)
 
         return sha1(key_json.encode()).hexdigest()  # noqa: S303 S324
+
+    def warn_property_not_found(  # noqa: C901
+        self,
+        rules: list[MetadataRule],
+        catalog: list[dict[str, t.Any]],
+    ):
+        """Validate MetadataRules conforms to discovered Catalog.
+
+        Validate MetadataRules against the tap's discovered Catalog & emit
+        warnings for each rule that does not match any property in catalog.
+
+        The method is intentionally written in a defensive manner so as not to
+        raise exceptions. e.g.
+            * use dict.get rather than []-accessor
+            * filter list comprehensions to remove some elements that should
+              not be used for validation
+
+        Args:
+            rules: List of `MetadataRule`
+            catalog: Discovered Source Catalog
+        """
+        property_rules = [
+            r
+            for r in rules
+            if r.tap_stream_id != "*"
+            and len(r.breadcrumb) >= 2
+            and r.breadcrumb != ["properties", "*"]
+        ]
+        stream_dict = {
+            stream.get("tap_stream_id"): stream
+            for stream in catalog.get("streams", [])
+            if isinstance(stream, dict)
+        }
+
+        def is_not_star(x):
+            return x != "*"
+
+        def dict_get(dictionary, key):
+            return dictionary.get(key, {})
+
+        for rule in property_rules:
+            if not (path := takewhile(is_not_star, rule.breadcrumb)):
```

Comment:
`takewhile` returns a _lazy_ iterator so I'm not sure which falsey value of `path` is expected to cause the code to follow this branch.

Reply:
you're correct, and as you have suggesting, wrapping in `list` would resolve this issue.